pandas/core/series.py

```diff
@@ -3272,7 +3274,13 @@ def combine_first(self, other) -> Series:
         if this.dtype.kind == "M" and other.dtype.kind != "M":
             other = to_datetime(other)
 
-        return this.where(notna(this), other)
+        combined = this.where(notna(this), other)
```

Comment:
Not sure if this won't break anything, but for the series case there might be a proper fix. You could set a compatible fill_value for the reindex ops. Only disadvantage is, that you'll have to compute ``notna(this)`` somehow